### PR TITLE
Added routes and data for users

### PR DESCRIPTION
**Pull request by troytomasch**
- Added data for users
- Added routes for users
- Added error handling functions for user fields

### CODE REVIEW

File: data/user.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+async function searchUsers(firstName, lastName) {
+  let fullName = [firstName, lastName];
+  errorHandler.checkIfElementsExists(fullName);
+  errorHandler.checkIfElementsAreStrings(fullName);
+  errorHandler.checkIfElementNotEmptyString(fullName);
+
+  fullName = [firstName.toLowerCase(), lastName.toLowerCase()];
+
+  const users = await userCollection();
```
Comment (tejasraibagi26) [inline data/user.js:77]: I think it should be 
```JavaScript
const users = await getAllUsers() 
```
else you'll be only holding the user collection in the variable.

**Review by tejasraibagi26 — CHANGES_REQUESTED** [review]

File: data/user.js (hunk 1 above)
Comment (tejasraibagi26) [reply data/user.js:77]: And also could you add a route for login via email?

Comment (troytomasch) [reply data/user.js]: Made these changes

File: routes/user.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    res.status(200).json(returnUser);
+  } catch (e) {
+    res.status(500).json({ err: e });
+  }
+});
+
+router.post("/login", async (req, res) => {
+  let username = req.body.email;
+  let inputPassword = req.body.password;
+
```
Comment (tejasraibagi26) [inline routes/user.js:106]: errorHandler for username and password missing.

**Review by tejasraibagi26 — CHANGES_REQUESTED** [review]

File: data/user.js (hunk 1 above)
Comment (tejasraibagi26) [reply data/user.js]: Everything looks great, just missing errorHandlers for /login route

**Review by tejasraibagi26 — APPROVED** [review]
Looks good, can be merged.

**Pull request merged**